rpc/rpc-transport/socket/src/name.c

```diff
@@ -357,9 +355,17 @@ af_inet_client_get_remote_sockaddr(rpc_transport_t *this,
         }
     }
 
+    // If transport-address family is already present in cmdline args, resolve
+    // right away. If family is set to AF_UNSPEC, we try and guess the family
+    if (sockaddr->sa_family == AF_INET || sockaddr->sa_family == AF_INET6) {
+        goto resolve;
+    }
+
     /* Need to update transport-address family if address-family is not provided
        to command-line arguments
     */
+    struct in6_addr serveraddr_ipv6;
+    struct in_addr serveraddr_ipv4;
```

Comment:
I wouldn't move that. In practice it doesn't make any difference (the compiler will still allocate the stack space at the beginning of the function), and from code readability, I think it's better to have all used variables in a single place.

Reply:
> the compiler will still allocate the stack space at the beginning of the function)

Got it. Thank you!